pd/migrations/20211216025603_init.sql

```diff
@@ -1,42 +1,55 @@
+-- Miscellaneous key-value storage for the application
```

Comment:
These changes to the schema are of three varieties:

1. extra comments, since this file is growing large enough that comments make it easier to quickly comprehend,
2. additional constraints as safety checks on the well-formedness of data we insert (some of these replace `TODO` comments calling for their creation), and
3. modifications to the quarantine tables to update them for the schema used in this work.

I wouldn't be opposed to splitting out (1) and (2) into separate PR(s), but I did them inline with this work, and if they seem acceptable, I think they are improvements we should merge.